sstables/sstables.cc

```diff
@@ -1260,6 +1260,38 @@ sstable::find_first_position_in_partition(reader_permit permit, const dht::decor
     co_return std::move(ret);
 }
 
+future<> sstable::load_first_and_last_position_in_partition() {
+    if (!_schema->clustering_key_size()) {
+        co_return;
+    }
+
+    try {
+        // Infinite timeout can cause silent lock-up, let's give the operation a generous time limit,
+        // as it can be fighting with thousands of sstables being loaded in parallel.
+#ifndef DEBUG
+        auto timeout = db::timeout_clock::now() + std::chrono::seconds(10);
+#else
+        // That's to prevent flakiness in debug-mode tests that can be super slow.
+        auto timeout = db::no_timeout;
+#endif
+        auto& sem = _manager.sstable_metadata_concurrency_sem();
+        reader_permit permit = co_await sem.obtain_permit(&*_schema, "sstable::load_first_and_last_position_range", sstable_buffer_size, timeout);
+        auto first_pos_opt = co_await find_first_position_in_partition(permit, get_first_decorated_key(), false);
+        auto last_pos_opt = co_await find_first_position_in_partition(permit, get_last_decorated_key(), true);
+
+        // Allow loading to proceed even if we were unable to load this metadata
+        if (!first_pos_opt || !last_pos_opt) {
+            sstlog.error("Unable to retrieve metadata for first and last keys of {}. Not a critical error.", get_filename());
+            co_return;
+        }
+
+        _first_partition_first_position = std::move(*first_pos_opt);
+        _last_partition_last_position = std::move(*last_pos_opt);
+    } catch (semaphore_timed_out&) {
```

Comment:
This is icky, because it introduces a new way to partially fail.

Why not use no_timeout exclusively?

Also @denesb reader_concurrency_semaphore should understand no_timeout as no limits on the internal queue.

Reply:
We can go back to using `no_timeout`, I have no strong opinion on the matter (even though I prefer defined timeouts).
No timeout makes the code easier, but in case of a problem the code can hang indefinitely. Shouldn't happen here (tm), I suggested using a timeout just in case...